controllers/oauth2.js

```diff
@@ -81,33 +81,50 @@ const displayScope = (scope, w) => {
 	}
 };
 
-router.get('/consent', csrfProtection, auth.authChecker, (r, w, next) => {
+router.get('/consent', csrfProtection, auth.authChecker, (req, res, next) => {
```

Comment:
You already refactore many stuff, it is possible that you transform this from promise chain to awaits? But please not that then arround the complet lines of code inside of the controller it must be added a try catch.